src/Components/Clock/Clock.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export class Clock extends React.Component {
+  state = {
+    date: (new Date()).toLocaleTimeString(),
+  }
+
+  componentDidMount() {
+    this.time = setInterval(() => {
+      this.setState({
+        date: (new Date()).toLocaleTimeString(),
+      });
+      // eslint-disable-next-line no-console
+      console.log(this.state.date);
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.clockName !== this.props.clockName) {
+      // eslint-disable-next-line no-console,max-len
+      console.log(`The Clock was renamed from ${prevProps.clockName} to ${this.props.clockName}`);
+    }
+  }
```

Comment:
It's great that you have used componentDidUpdate, but for this task, there's no need for it, you should only console.log the new name only when clicking the button (so you could easily put the 'console.log' right directly after giving a new name)

Reply:
What's wrong with componentDidUpdate? If I don't use this method then I have to store the old name somewhere and componentDidUpdate gets prevProps, I think this method is perfect for solving this problem.

Reply:
Of course, it's up to you. As I've told you earlier - it's great, that you have used it. But my point is that for simple logging into the console after renaming there would be more meaningful to do it right away where you changing the name. Now it's hard to understand that's renaming the clock is that thing that triggers logging into the console the new name. And if there be some other functionalities in this task, that would re-render component, every time this check will be called

Reply:
I understood your idea, thanks for the explanation, I will take note